src/resources/ResourceSnapshots/ResourceSnapshots.ts

```diff
@@ -45,19 +45,52 @@ export default class ResourceSnapshots extends Resource {
         return await fetch(url, {method: 'get'});
     }
 
-    createFromFile(file: File, options: CreateFromFileOptions) {
-        const computedOptions = {developerNotes: options.developerNotes, snapshotFileType: undefined};
+    /**
+     * Creates a snapshot from a file containing the configuration.
+     *
+     * @param {File} file The file containing the configuration.
+     * @param {CreateFromFileOptions} options
+     */
+    createFromFile(file: File, options: CreateFromFileOptions): Promise<ResourceSnapshotsModel>;
+
+    /**
+     * Creates a snapshot from a file buffer containing the configuration.
+     *
+     * @param {Buffer} file The file containing the configuration.
+     * @param {ResourceSnapshotSupportedFileTypes} fileType The type of the file containing the configuration.
+     * @param {CreateFromFileOptions} options
+     */
+    createFromFile(
+        file: Buffer,
+        fileType: ResourceSnapshotSupportedFileTypes,
```

Comment:
The second parameter has a different type in the 2 signatures, so I don't see what name we can give to make it clear. :S

Reply:
Instead of overriding `createFromFile` , could we leave the old method signature exactly as is, and add a `createFromBuffer(buf: Buffer, fileType: ResourceSnapshotSupportedFileTypes, options: CreateFromFileOptions)` function ?

It should simplify usage.

And then just extract the shared code for both function in an internal/private function.

Reply:
> Instead of overriding `createFromFile` , could we leave the old method signature exactly as is, and add a `createFromBuffer(buf: Buffer, fileType: ResourceSnapshotSupportedFileTypes, options: CreateFromFileOptions)` function ?
> 
> It should simplify usage.
> 
> And then just extract the shared code for both functions in an internal/private function.

GIven that the first method is named `createFromFIle` that could make sense.
I would have prefer to have 'one API route, one method' tbh, but it'll end up pretty much the same here, and the signature/typings would prolly be more user-friendly they way you proposes.

Reply:
I agree that the overload is kinda messy. But I also prefer a 1:1 relationship between the API and the methods tbh.
A Node user would probably try to use `createFromFile` before realizing that the File API is not supported by Node and then look for a different method.
As opposed to see the 2 available signatures when using the method.

Reply:
Alright !